packages/web/docs/src/pages/docs/gateway/other-features/plugins.mdx

```diff
@@ -0,0 +1,628 @@
+Envelop plugin system by providing gateway specific hooks.
+
+This means that if a plugin exists in multiple variants, you should follow this priority order:
+
+1.  Hive Gateway: is aware of subgraphs and upstream request management.
+2.  Yoga: is aware of the HTTP transport. It most of the time provides optimisations over the
+    Envelop variants.
+3.  Envelop: is only aware of the GraphQL operation execution.
+
+For example, the Prometheus integration plugin is available for Envelop (`@envelop/prometheus`),
+Yoga(`@graphql-yoga/plugin-prometheus`) and Hive Gateway(`@graphql-mesh/prometheus`). In this case,
+you should pick `@graphql-mesh/prometheus` plugin. It is built on top of Yoga and Envelop variants,
+but adds monitoring capabilities of upstream subgraphs.
+
+For the most used features, you will not have to explicitly setup the dedicated plugin. They will be
+added automatically based on your gateway configuration.
+
+## Configuration
+
+You can add plugins to your gateway using the `plugins` option.
+
+It have to be a function, which will be called each time the gateway have update its configuration.
+For example, if polling is enabled, this function will be called for each poll.
+
+Most Hive Gateway plugins takes an object as a parameter, and expect some common components like a
+`logger`, a `pubsub`, etc... Those components are given in parameters to the `plugins` function. It
+is advised to spread the plugin's factory context into the plugins options, this way plugins will
+have access to all components they need.
+
+```ts filename="gateway.config.ts"
+import { useGraphQLJit } from '@envelop/graphql-jit'
+import { defineConfig } from '@graphql-hive/gateway'
+import useNewRelic from '@graphql-mesh/plugin-newrelic'
+
+export const gatewayConfig = defineConfig({
+  plugins: ctx => [
+    useGraphQLJit(),
+    useNewRelic({
+      ...ctx
+      // other options
+    })
+  ]
+})
+```
+
+## Custom Plugins
+
+You can take advantage of the plugin system to create your own custom plugin. This allows you to
+integrate your gateway with any technology that we don't officially support, being public or private
+to your company.
+
+A good entry-point for discovering how to write Gateway plugins is to look at the source code of the
+existing plugins maintained by us.
+
+A plugin is an object that can define methods to hook into different phases or events of the gateway
+lifecycle. Each hook will receive a single parameter, the hook payload, containing data related to
+the hooked phase and functions allowing you to modify the behavior of this phase.
+
+There are 2 categories of hooks:
+
+- Gateway lifecycle hooks: they are related to the lifecycle or events of the server itself and is
+  not related to a specific request.
+- Request lifecycle hooks: they are tight to a specific phase of a specific request. Those hooks
+  always give access to the related request in there payload.
```

Comment:
I think a mermid chart of the hooks would be nice here

Reply:
Like the one below in 184th line?

Reply:
It is possible to see with `View File`;
https://github.com/graphql-hive/console/blob/3ed148ed148ef8677be8e4d927e35fd1951be88b/packages/web/docs/src/pages/docs/gateway/other-features/plugins.mdx
GitHub is able to show mermaid graphs.

Reply:
There is a mermaid chart for the Request lifecycle. I was not sure yet if a Gateway lifecycle graph would be helpfull, but I can make one if you think so

Reply:
I've added a diagram for the gateway lifecycle, let me know what you think about it